include/picongpu/plugins/saxs/Saxs.kernel

```diff
@@ -0,0 +1,303 @@
+                        if (linearIdx < particlesInFrame)
+                        {
+
+                            auto par = frame[linearIdx];
+
+                            // calculate global position
+                            lcellId_t const cellIdx = par[localCellIdx_];
+
+                            // position inside of the cell
+                            floatD_X const pos = par[position_];
+
+                            // calculate global position of cell
+                            DataSpace<simDim> const globalPos(
+                                superCellOffset +
+                                DataSpaceOperations<simDim>::
+                                    template map<SuperCellSize>(cellIdx));
+
+                            // add global position of cell with local position of particle in cell
+                            // set z component to zero in case of simDim==DIM2
+                            r_s[linearIdx][2] = 0.0;
+                            // run over all components and compute gobal position
+                            for (int i = 0; i < simDim; ++i)
+                                r_s[linearIdx][i] =
+                                    (float_X(globalPos[i]) + pos[i]) *
+                                    cellSize[i];
+
+                            /* get macro-particle weighting
+                                         *
+                                         * Info:
+                                         * the weighting is the number of real particles described
+                                         * by a macro-particle
+                                         */
+                            saxsWeighting_s[linearIdx] = par[weighting_];
+
+                        } // END: only threads with particles are running
+                    }
+                );
+
+                __syncthreads(); // wait till every thread has loaded its particle data
+
+                // run over all q for this thread
+                for (int idx = workerIdx; idx < n_q; idx += T_numWorkers)
+                {
+
+                    // Initialize in register memory
+                    float1_64 sumfcoskr_r = 0;
+                    float1_64 sumfsinkr_r = 0;
+                    float3_64 q_r;
+                    float_64 dotkr;
+
+                    int i_z = idx % n_qz;
+                    int i_y = (idx / n_qz) % n_qy;
+                    int i_x = idx / (n_qz * n_qy);
+
+                    q_r[0] = (q_min[0] + q_step[0] * i_x);
+                    q_r[1] = (q_min[1] + q_step[1] * i_y);
+                    q_r[2] = (q_min[2] + q_step[2] * i_z);
+
+                    /// Particle loop: thread runs through loaded particle data
+                    for (int j = 0; j < particlesInFrame; ++j)
+                    {
+                        dotkr = (q_r[0] * r_s[j][0] + q_r[1] * r_s[j][1] + q_r[2] * r_s[j][2])*1e10*UNIT_LENGTH; //in PIConGPU unit
```

Comment:
`1e10` is a magic constant and of type `int`.
please store it in a temporary `const` variable to explain it (probably something in Ångström?)

Reply:
Defined a const variable `float_64 const meter2angstrom = 1e10;`